app_feup/lib/view/Widgets/generic_card.dart

```diff
@@ -72,7 +75,9 @@ class GenericCardState extends State<GenericCard> {
                 ),
                 child: Container(
                   decoration: BoxDecoration(
-                      color: Theme.of(context).primaryColor,
+                      color: Theme.of(context).brightness == Brightness.light
+                          ? Colors.white
+                          : Colors.black,
```

Comment:
If you accept the cardColor sugestion, this will be just one liner

```suggestion
                      color: Theme.of(context).cardColor,
```